### PR TITLE
[Data] Add Vale configuration

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

This PR adds configuration for Vale -- a linting tool that helps us enforce a consistent editorial style. To use Vale, either install the [VSCode extension for Vale](https://marketplace.visualstudio.com/items?itemName=errata-ai.vale-server) or run `vale doc/source/data`.

In a future PR, we plan to enforce Vale in CI (starting with the Data documentation).

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by amogkam** [comment]
Nice! How does this work exactly? How does this affect Ray contributors?

File: .github/styles/Google/We.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+extends: existence
+message: "Try to avoid using first-person plural like '%s'."
+link: 'https://developers.google.com/style/pronouns#personal-pronouns'
+level: warning
+ignorecase: true
+tokens:
+  - we
+  - we'(?:ve|re)
+  - ours?
```
Comment (angelinalg) [inline .github/styles/Google/We.yml:9]: does this include "our"?

File: .github/styles/Vocab/General/reject.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+[Pp]lease
```
Comment (angelinalg) [inline .github/styles/Vocab/General/reject.txt:1]: THANK YOU!!!!!!
Also, "congratulations."

File: .github/styles/Google/Will.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+extends: existence
+message: "Avoid using '%s'."
+link: 'https://developers.google.com/style/tense'
+ignorecase: true
+level: warning
+tokens:
+  - will
```
Comment (angelinalg) [inline .github/styles/Google/Will.yml:7]: How about `'ll`

**Comment by angelinalg** [comment]
This literally gives me the warm fuzzies. Thank you so much, @bveeramani and @amogkam!

One request, @bveeramani, is there a way to standardize how we tell readers to "see" another doc, vs "read, go to, check out, etc."

File: .github/styles/Google/Acronyms.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  - CPU
+  - CSS
+  - CSV
+  - CUDA
+  - DEBUG
+  - DOM
+  - DPI
+  - ETL
+  - FAQ
+  - GCC
```
Comment (angelinalg) [inline .github/styles/Google/Acronyms.yml:24]: This may be a stupid comment, but how about GCP and GCE?

**Review by angelinalg — APPROVED** [review]
Just some nits.

**Comment by amogkam** [comment]
All credit goes to @bveeramani! Great work!

File: .github/styles/Google/EnDash.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+extends: existence
+message: "Use an em dash ('—') instead of '–'."
```
Comment (amogkam) [inline .github/styles/Google/EnDash.yml:2]: these 2 dashes look the same to me...how does one even do an em dash?

**Review by amogkam — APPROVED** [review]
data doc changes lgtm

File: .github/styles/Google/EnDash.yml (hunk 5 above)
Comment (bveeramani) [reply .github/styles/Google/EnDash.yml:2]: They look the same in monospace, but they look different in other typefaces.

image

@angelinalg what're your thoughts on this rule?

File: .github/styles/Google/Acronyms.yml (hunk 4 above)
Comment (bveeramani) [reply .github/styles/Google/Acronyms.yml:24]: Ah, good catch. Looks like all our uses of "GCP" and "GCE" are in special blocks like dropdowns and tabs. Those widgets aren't linted due to a limitation of Vale.

Fixed.

File: .github/styles/Google/We.yml (hunk 1 above)
Comment (bveeramani) [reply .github/styles/Google/We.yml:9]: Yeah, the "?" means the "s" is optional.

File: .github/styles/Google/EnDash.yml (hunk 5 above)
Comment (bveeramani) [reply .github/styles/Google/EnDash.yml:2]: > how does one even do an em dash

Also @amogkam to answer your question, I just google "em dash" and then copy it lol

image

Comment (bveeramani) [reply .github/styles/Google/EnDash.yml]: Okay, did some more research. Sphinx renders two normal dashes (`--`) as a en dash (–) and three normal dashes (`---`) as an em dash (—). 

I've updated the Vale rules accordingly.

**Pull request merged**